src/page/Themes/hooks/useIntersectionObserver.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useRef } from 'react';
+
+interface useIntersectionObserverProps {
+  onIntersect: () => void;
+  enable: boolean;
+}
+
+const useIntersectionObserver = ({ onIntersect, enable = true }: useIntersectionObserverProps) => {
+  const ref = useRef(null);
+
+  useEffect(() => {
+    if (!enable) {
+      return;
+    }
+    const observer = new IntersectionObserver(
+      async ([entry]) => {
+        if (entry.isIntersecting) {
+          onIntersect();
+        }
+      },
+      { threshold: 1.0 }
+    );
+
+    const el = ref.current;
+    if (el) observer.observe(el);
+
+    return () => {
+      if (el) observer.unobserve(el);
+    };
+  }, [enable, onIntersect]);
```

Comment:
```suggestion
  useEffect(() => {
    const el = ref.current;
    if (!enable || !el) {
      return;
    }
    const observer = new IntersectionObserver(
      async ([entry]) => {
        if (entry.isIntersecting) {
          onIntersect();
        }
      },
      { threshold: 1.0 }
    );

    observer.observe(el);

    return () => observer.unobserve(el);
  }, [enable, onIntersect]);
```

이렇게 아예 상단에서 막아버릴 수 있답니다 ㅎㅎ